subcommands/keys/common.go

```diff
@@ -25,45 +26,142 @@ import (
 type OfflineCreds map[string][]byte
 
 type TufSigner struct {
-	Id  string
-	Key *rsa.PrivateKey
+	Id      string
+	SigType string
+	Key     crypto.Signer
 }
 
 type keypair struct {
-	rsaPriv      *rsa.PrivateKey
+	signer       TufSigner
 	atsPriv      client.AtsKey
 	atsPrivBytes []byte
+	atsPub       client.AtsKey
+	atsPubBytes  []byte
+}
+
+type TufKeyType interface {
+	Name() string
+	SigName() string
+	GenerateKey() (crypto.Signer, error)
+	ParseKey([]byte) (crypto.Signer, error)
+}
+
+type tufKeyTypeRSA struct{}
+type tufKeyTypeEd25519 struct{}
+
+const (
+	pemPrivateKeySuffix = " PRIVATE KEY"
+	// These are case insensitive
+	tufKeyTypeNameEd25519    = "ED25519"
+	tufKeyTypeNameRSA        = "RSA"
+	tufKeyTypeSigNameEd25519 = "ed25519"
+	tufKeyTypeSigNameRSA     = "rsassa-pss-sha256"
+)
+
+var (
+	gtufKeyTypeEd25519 = &tufKeyTypeEd25519{}
```

Comment:
annoying how golang makes you do this. i just went to suggest 3 different alternatives but they are probably more annoying.

Reply:
yeah... I am quite annoyed that Golang doesn't allow to cast inline structs to an interface... they need a pointer in memory.